management/networksecuritygroup/entities.go

```diff
@@ -0,0 +1,45 @@
+// Package networksecuritygroup implements operations for managing network security groups
+// using the Service Management REST API
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn913824.aspx
+package networksecuritygroup
+
+import (
+	"encoding/xml"
+
+	"github.com/MSOpenTech/azure-sdk-for-go/management"
+)
+
+// SecurityGroupClient is used to manage operations on network security groups
+type SecurityGroupClient struct {
+	client management.Client
+}
+
+// SecurityGroup represents a network security group
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn913821.aspx
+type SecurityGroup struct {
```

Comment:
Yeah, I must admit I was still looking at the structs/entities so will tweak them a little... This PR is mainly here because I wanted to open the discussion on styles and approaches, so will do some refactoring on these before coming to a final (squashed) PR.